### DISCUSSION TITLE
If you could rebuild this library from scratch...

### DISCUSSION THREAD

**Discussion by valtism** [airbnb/visx#1416] [General]
Hi there, question for our lovely owners/maintainers.

I'm thinking of attempting to write a port of this library for [SolidJS](https://www.solidjs.com). Since I'll be basically writing from scratch, I was wondering if there were any API design choices made that you wish were different? Things you cannot change now because they would be too difficult, or would break the API too much.

Any thoughts or opinions around this as a whole are very much welcome.

**Comment by Stuhl** [comment]
Even though I'm not a maintainer of the library I would say make every component a controlled component. Meaning that the components should expose ways to force updates for instance. It's difficult to implement certain features if you have to do weird work arounds for things that aren't public to the client.

The Brush for instance is a example where some parts are uncontrolled and some controlled.

Also make sure to include a real tutorial. The "tutorials" here are not beginner friendly.
